message_ix/utils.py

```diff
@@ -16,6 +21,105 @@ def logger():
     return _LOGGER
 
 
+def isstr(x):
+    """Returns True if x is a string"""
+    return isinstance(x, six.string_types)
+
+
+def isscalar(x):
+    """Returns True if x is a scalar"""
+    return not isinstance(x, collections.Iterable) or isstr(x)
+
+
+def add_spatial_sets(scenario, data):
```

Comment:
I think that this syntax will lead to confusion in the medium term - why is it `scenario.function(...)` in some cases and `message_ix.function(scenario, ...)` in others?

Reply:
Possible, but this is the same as in Pandas. Other options exist, like subclassing `Scenario`, make a `ScenarioBuilder` object, etc. 

For this function, it may even be possible to just add it directly to `ixmp`, as I don't think there's anything message specific here. Or is there?

Reply:
The more I think about it, the more I like subclassing..